NOTES.WIN

```diff
@@ -83,41 +102,60 @@
  is, of course, to choose a different set of directories by using
  --prefix and --openssldir when configuring.
 
-
  Special notes for Universal Windows Platform builds, a.k.a. VC-*-UWP
  --------------------------------------------------------------------
 
  - UWP targets only support building the static and dynamic libraries.
 
  - You should define the platform type to "uwp" and the target arch via
    "vcvarsall.bat" before you compile. For example, if you want to build
-   "arm64" builds, you should type "vcvarsall.bat x86_arm64 uwp".
+   "arm64" builds, you should run "vcvarsall.bat x86_arm64 uwp".
 
- mingw and mingw64
- =================
 
- * MSYS2 shell and development environment installation:
+ Native OpenSSL built using MinGW
+ ================================
 
-   Download MSYS2 from https://msys2.github.io/ and follow installation
-   instructions. Once up and running install even make, perl, (git if
-   needed,) mingw-w64-i686-gcc and/or mingw-w64-x86_64-gcc. You should
-   have corresponding MinGW items on your start menu, use *them*, not
-   generic MSYS2. As implied in opening note, difference between them
-   is which compiler is found 1st on $PATH. At this point ./config
-   should recognize correct target, roll as if it was Unix...
+ MinGW offers an alternative way to build native OpenSSL, by cross compilation.
 
- * It is also possible to build mingw[64] on Linux or Cygwin by
-   configuring with corresponding --cross-compile-prefix= option. For
-   example
+ * Usually the build is done on Windows in a GNU-like environment called MSYS2.
 
-     ./Configure mingw --cross-compile-prefix=i686-w64-mingw32- ...
+   MSYS2 provides GNU tools, a Unix-like command prompt,
+   and a UNIX compatibility layer for applications.
+   However in this context it is only used for building OpenSSL.
+   The resulting OpenSSL does not rely on MSYS2 to run and is fully native.
+
+   Requirement details
+
+   - MSYS2 shell, from https://www.msys2.org/
+
+   - MinGW[64] compiler: mingw-w64-i686-gcc and/or mingw-w64-x86_64-gcc.
+     These compilers must be on your MSYS2 $PATH.
+     A common error is to not have these on your $PATH.
+     The MSYS2 version of gcc will not work correctly here.
+
+   - Perl, at least version 5.10.0, installed into the MSYS2 environment
+
+   - make, installed into the MSYS2 environment
+
+   In the MSYS2 shell do the configuration depending on the target architecture:
 
+     ./Configure mingw ...
    or
+     ./Configure mingw64 ...
```

Comment:
You should just be able to do `./config` here. It *should* auto-detect the environment correctly (but I don't have easy access to a Windows environment at the moment to confirm its still working...my Windows environment died).

Reply:
I can confirm that it does

Reply:
> You should just be able to do ./config here. It should auto-detect the environment correctly 

Yes, it picks the default architecture (likely depending on the MSYS2 variant installed).
I've added this as a third option.

> I don't have easy access to a Windows environment at the moment to confirm its still working...my Windows environment died

Mine is also a bit crippled these days. I have not (yet re-)installed MSVC after the last fresh system installation, but the configuration scripts can be tried out regardless.

> I can confirm that it does

Me too. Yet strangely with MSYS2 the configuration takes several minutes here.
Also other perl and make runs (e.g., for converting the POD files) are painfully slow.
Looks like there is an issue with the MSYS2 file system access, which may be a peculiarity with my company's Windows additions like the Trend Micro virus scanner.